cmake/SearchForStuff.cmake

```diff
@@ -326,6 +326,22 @@ if (PKG_CONFIG_FOUND)
     endif (NOT TBB_FOUND)
   endif (NOT TBB_FOUND)
 
+  # tbb::task was removed in tbb v2021.01, so in this case we must move to the
+  # newer tbb::task_group API. This could also be used with earlier versions of
+  # TBB, but we want to maintain ABI compatibility for Ubuntu 18.04 etc., so
+  # disable this by default.
+  if (NOT DEFINED USE_LEGACY_TBB_TASK)
+    if (${TBB_VERSION} VERSION_LESS 2021)
+      set (USE_LEGACY_TBB_TASK ON)
+    else (${TBB_VERSION} VERSION_LESS 2021)
+      set (USE_LEGACY_TBB_TASK OFF)
+    endif(${TBB_VERSION} VERSION_LESS 2021)
+  endif()
+  set(USE_LEGACY_TBB_TASK ${USE_LEGACY_TBB_TASK} CACHE BOOL "Whether to use the deprecated tbb::task class")
+  if (USE_LEGACY_TBB_TASK)
+    add_definitions(-DUSE_LEGACY_TBB_TASK)
```

Comment:
This definition needs to be propagated to downstream compilation units as it is used in public headers, but unfortunately gazebo does not use modern style CMake targets so there is no straightforward way to do so. I can  check if there is any facility currently use to expose definitions to downstream projects, like a `GAZEBO_DEFINITIONS` CMake variable.

Reply:
This sounds like a more sensible approach. I'll have a go :smile:

Reply:
Thanks!